lib/common/components/Login.js

```diff
@@ -1,31 +1,35 @@
 // @flow
 
-import {Component} from 'react'
+import { useAuth0 } from '@auth0/auth0-react'
+// $FlowFixMe useEffect not recognized by flow.
+import { useEffect } from 'react'
+import { browserHistory } from 'react-router'
 
-import * as userActions from '../../manager/actions/user'
-import auth0 from '../user/Auth0Manager'
-
-import type {Props as ContainerProps} from '../containers/Login'
-
-type Props = ContainerProps & {
-  onHide: () => void,
-  push: string => void,
-  receiveTokenAndProfile: typeof userActions.receiveTokenAndProfile,
-  redirectOnSuccess: string
+type Props = {
+  redirectOnSuccess?: string
 }
 
-export default class Auth0 extends Component<Props> {
-  componentDidMount () {
-    // Show lock when component mounts. NOTE: render method returns null.
-    auth0.loginWithLock(this.props)
-  }
-
-  componentWillUnmount () {
-    // Hide lock when component unmounts
-    auth0.hideLock()
-  }
+export default function Login ({
+  redirectOnSuccess = window.location.path || '/home'
+}: Props) {
+  const auth0 = useAuth0()
+  const { isAuthenticated, isLoading } = auth0
+  useEffect(() => {
+    if (!isLoading && redirectOnSuccess) {
+      if (!isAuthenticated) {
+        // Show the login page if the user state is not logged in.
+        auth0.loginWithRedirect({
+          appState: {
+            returnTo: redirectOnSuccess
+          }
+        })
+      } else {
+        // Otherwise just go to the target url.
+        browserHistory.push(redirectOnSuccess)
+      }
+    }
+  }, [isAuthenticated, isLoading, redirectOnSuccess])
```

Comment:
Actually the purpose of this component is to redirect ASAP, so nothing happens (and nothing is rendered) until variables change so that the redirect condition is met.